src/commands/convert.js

```diff
@@ -16,64 +16,120 @@ const convert = (context, appid, location) => {
+      }/api/platform/cli/apps/${appid}/versions/${context.argOpts.version}`;
 
-    utils.startSpinner('Downloading app from Zapier');
+      utils.startSpinner('Downloading app from Zapier');
+      let [appInfo, versionInfo] = [];
+      try {
+        [appInfo, versionInfo] = await Promise.all([
+          utils.callAPI(null, { url: appInfoUrl }, true),
+          utils.callAPI(null, { url: versionInfoUrl }, true)
+        ]);
+      } catch (e) {
+        if (e.status === 404) {
+          throw new Error(
+            `Visual Builder app ${appid} @ ${
+              context.argOpts.version
+            } not found. If you want to convert a Legacy Web Builder app, don't pass a \`--version\` option.`
+          );
+        }
+      }
+      utils.endSpinner();
+      return utils.convertVisualApp(appInfo, versionInfo, tempAppDir);
+    } else {
+      // has info about the app, such as title
+      const legacyDumpUrl = `${
+        constants.BASE_ENDPOINT
+      }/api/developer/v1/apps/${appid}/dump`;
+      // has a CLI version of the actual app implementation
+      const cliDumpUrl = `${
+        constants.BASE_ENDPOINT
+      }/api/developer/v1/apps/${appid}/cli-dump`;
 
-    const [legacyApp, appDefinition] = await Promise.all([
-      utils.callAPI(null, { url: legacyDumpUrl }),
-      utils.callAPI(null, { url: cliDumpUrl })
-    ]);
+      utils.startSpinner('Downloading app from Zapier');
+      let [legacyApp, appDefinition] = [];
+      try {
+        [legacyApp, appDefinition] = await Promise.all([
+          utils.callAPI(null, { url: legacyDumpUrl }, true),
+          utils.callAPI(null, { url: cliDumpUrl }, true)
+        ]);
+      } catch (e) {
+        if (e.status === 404) {
+          throw new Error(
+            `Legacy Web Builder app ${appid} not found. If you want to convert a Visual Builder app, make sure to pass a \`--version\` option.`
+          );
+        }
+      }
 
-    // The JSON dump of the app doesn't have app ID, let's add it here
-    legacyApp.general.app_id = appid;
+      // The JSON dump of the app doesn't have app ID, let's add it here
+      legacyApp.general.app_id = appid;
 
-    utils.endSpinner();
+      utils.endSpinner();
 
-    return await utils.convertApp(legacyApp, appDefinition, tempAppDir);
+      return utils.convertLegacyApp(legacyApp, appDefinition, tempAppDir);
+    }
   };
 
   return utils.initApp(context, location, createApp).then(() => {
     context.line();
     context.line(
-      `Finished! You may try \`npm install\` and then \`zapier test\` in "${location}" directory.`
+      `Finished! You may try \`npm install\` and then \`zapier test\` in the "${location}" directory.`
     );
     context.line(
       "Also, if your app has authentication, don't forget to set environment variables:"
     );
     context.line('* for local testing, edit the .env file');
-    context.line('* for produciton, use `zapier env` command');
+    if (!isVisual) {
```

Comment:
I think I agree. The biggest upside we get as written is that it's not a breaking change and can be released sooner (originally we promised that we would only do breaking changes every so often). 

So I think we have a few options here: 

1. leave it as is, which forces the dev to be explicit about which version they want
2. provide the `--latest` flag, which needs to be provided instead of the `--version` flag. Either denotes the command as referring to a Godzila app. In a breaking change, we could require a flag to explicitly type the app (`--visual` or `--legacy`)
3. Do a more full featured picker (similar to `zapier link`). 

Do you have strong feelings here? I feel like the UX as written is "good enough", especially since devs wont be running this often.